src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/InfiniteScrollList.xaml.cs

```diff
@@ -143,9 +161,29 @@ private int FilteredItemsCount
 
         public Guid? OperationId => _loader?.State.OperationId;
 
-        public int TopLevelPackageCount { get; private set; }
+        public int TopLevelPackageCount
+        {
+            get
+            {
+                return FilterByVulnerabilities is true ? TopLevelVulnerablePackagesCount : TopLevelAllPackagesCount;
```

Comment:
Do we need separate properties? If we're applied the filter, why do we need the separate counts?

Reply:
This counts are for the collapsers/separators of TopLevel and Transitive's, sadly I'm not able to access the group lengths 😢. With these counters I'm updating how many packages are in each group depending on the Checkbox status